bin/uniref90_non_viral_filter.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import gzip
+
+from Bio import SeqIO, Entrez
+import taxoniq
+
+
+def is_virus_taxoniq(tax_id):
+    """
+    Check if taxa id is virus using faster taxoniq search in local, 
+    indexed, compressed copy of the NCBI taxonomy database
+    """
+    taxon = taxoniq.Taxon(tax_id)
+    if taxon.ranked_lineage:
+        highest_taxon = taxon.ranked_lineage[-1]
+        if highest_taxon.rank.name == "superkingdom":
+            return highest_taxon.scientific_name == "Viruses"
+    raise ValueError("No information about the lineage in taxoniq")
+
+
+def is_virus_entrez(taxid):
+    """
+    Check if taxa id is virus using slower approach with
+    Entrez quering
+    """
+    handle = Entrez.efetch(db="taxonomy", id=taxid, retmode="xml")
+    records = Entrez.read(handle)
+    handle.close()
+
+    lineage = records[0]["Lineage"]
+    highest_taxon = lineage.split("; ")[0]
+    return highest_taxon == "Viruses"
+
+
+def filter_fasta(in_handle, out_handle, err_handle):
+    """
+    Filter viral proteins from FASTA using TaxID field.
+    """
+    for record in SeqIO.parse(in_handle, "fasta"):
+        tax_id = record.description.split("TaxID=")[1].split()[0]
```

Comment:
```suggestion
        tax_id = record.description.split("TaxID=")[1].split()[0]
        is_viral_protein = False
```

Reply:
I don't see any difference 😅
Was this made by accident?

Reply:
he added definition of `is_viral_protein = False` because it was not ddefined before try-except :think_about_it:

Reply:
Oh, that's true 😅
Thanks @KateSakharova

Reply:
Yeah, I think Python is okay with that. It's an old habit of mine to define variables in a wider scope.